src/test/java/io/orkes/conductor/client/e2e/JDBCWorkerTests.java

```diff
@@ -0,0 +1,347 @@
+            Workflow workflow = workflowClient.getWorkflow(workflowId, true);
+            assertTrue(workflow.getTasks().get(0).getStatus().isTerminal());
+        });
+
+        Workflow workflow = workflowClient.getWorkflow(workflowId, true);
+        Task task = workflow.getTasks().get(0);
+
+        assertEquals(Task.Status.COMPLETED, task.getStatus());
+        assertEquals(10, ((List<Object>) task.getOutputData().get("result")).size());
+    }
+
+    @Test
+    public void testJDBCWorkerUpdate() {
+        String query = "UPDATE " + TABLE_NAME +
+                "   SET order_status = 'Pending'" +
+                "   WHERE order_status = 'Shipped'";
+        Map<String, Object> inputMap = Map.of(
+                "connectionId", "postgres1",
+                "statement", query,
+                "parameters", List.of(),
+                "type", "UPDATE"
+        );
+
+        StartWorkflowRequest startWorkflowRequest = new StartWorkflowRequest();
+        startWorkflowRequest.setName(WORKFLOW_NAME);
+        startWorkflowRequest.setInput(inputMap);
+
+        String workflowId = workflowClient.startWorkflow(startWorkflowRequest);
+
+        await().atMost(30, TimeUnit.SECONDS).untilAsserted(() -> {
+            Workflow workflow = workflowClient.getWorkflow(workflowId, true);
+            assertTrue(workflow.getTasks().get(0).getStatus().isTerminal());
+        });
+
+        Workflow workflow = workflowClient.getWorkflow(workflowId, true);
+        Task task = workflow.getTasks().get(0);
+
+        assertEquals(Task.Status.COMPLETED, task.getStatus());
+        assertEquals(5, (Long) task.getOutputData().get("update_count"));
+    }
+
+    @Test
+    public void testJDBCWorkerUpdateWithExpectedCountSuccess() {
+        String query = "UPDATE " + TABLE_NAME +
+                "   SET order_status = ?" +
+                "   WHERE order_status = ?";
+        Map<String, Object> inputMap = Map.of(
+                "connectionId", "postgres1",
+                "statement", query,
+                "parameters", List.of("Shipped", "Pending"),
+                "type", "UPDATE",
+                "expectedUpdateCount", "3"
+        );
+
+        StartWorkflowRequest startWorkflowRequest = new StartWorkflowRequest();
+        startWorkflowRequest.setName(WORKFLOW_NAME);
+        startWorkflowRequest.setInput(inputMap);
+
+        String workflowId = workflowClient.startWorkflow(startWorkflowRequest);
+
+        await().atMost(30, TimeUnit.SECONDS).untilAsserted(() -> {
+            Workflow workflow = workflowClient.getWorkflow(workflowId, true);
+            assertTrue(workflow.getTasks().get(0).getStatus().isTerminal());
+        });
+
+        Workflow workflow = workflowClient.getWorkflow(workflowId, true);
+        Task task = workflow.getTasks().get(0);
+
+        assertEquals(Task.Status.COMPLETED, task.getStatus());
+        assertEquals(3, (Long) task.getOutputData().get("update_count"));
```

Comment:
How are three rows getting updated?

Reply:
During the insertion of data into the table, I inserted 3 records having `order_status` = `Pending`. So, in the update operation, it must update 3 records.